database/rrdset.c

```diff
@@ -525,108 +525,111 @@ RRDSET *rrdset_create_custom(
+        } else {
+            if (module != st->module_name) {
+                if (st->module_name)
+                    old_module = st->module_name;
+                st->module_name = module ? strdupz(module) : NULL;
+                mark_rebuild = 1;
             }
+        }
 
-            RRDSET_TYPE new_chart_type =
-                rrdset_type_id(config_get(st->config_section, "chart type", rrdset_type_name(chart_type)));
-            if (st->chart_type != new_chart_type) {
-                st->chart_type = new_chart_type;
+        char *new_title = title ? strdupz(title) : NULL;
+        if (new_title) {
+            json_fix_string(new_title);
+            if (unlikely(strcmp(st->title, new_title))) {
+                old_title = st->title;
+                st->title = new_title;
                 mark_rebuild = 1;
-            }
+            } else
+                freez(new_title);
+        }
 
-            char *new_family = family ? strdup(family): NULL;
-            if (new_family) {
-                json_fix_string(new_family);
-                if (unlikely((strcmp(st->family, new_family)))) {
-                    freez(st->family);
-                    st->family = new_family;
-                    mark_rebuild = 1;
-                } else
-                    freez(new_family);
-            }
+        RRDSET_TYPE new_chart_type =
+            rrdset_type_id(config_get(st->config_section, "chart type", rrdset_type_name(chart_type)));
+        if (st->chart_type != new_chart_type) {
+            st->chart_type = new_chart_type;
+            mark_rebuild = 1;
+        }
 
-            char *new_context = context ? strdup(context) : NULL;
-            if (new_context) {
-                json_fix_string(new_context);
-                if (unlikely((strcmp(st->context, new_context)))) {
-                    freez(st->context);
-                    st->context = new_context;
-                    st->hash_context = simple_hash(st->context);
-                    mark_rebuild = 1;
-                } else
-                    freez(new_context);
-            }
+        char *new_family = family ? strdup(family): NULL;
+        if (new_family) {
+            json_fix_string(new_family);
+            if (unlikely((strcmp(st->family, new_family)))) {
+                old_family = st->family;
+                st->family = new_family;
+                mark_rebuild = 1;
+            } else
+                freez(new_family);
+        }
 
-            rrdhost_unlock(host);
+        char *new_context = context ? strdup(context) : NULL;
+        if (new_context) {
+            json_fix_string(new_context);
+            if (unlikely((strcmp(st->context, new_context)))) {
+                old_context = st->context;
+                st->context = new_context;
+                st->hash_context = simple_hash(st->context);
+                mark_rebuild = 1;
+            } else
+                freez(new_context);
+        }
 
+        if (mark_rebuild) {
```

Comment:
`if (mark_rebuild && (old_plugin || old_module ...))`

Reply:
We don't want the pause if the collector sent the CHART again but nothing changed.